dbms/src/TiDB/Schema/TiDBSchemaSyncer.h

```diff
@@ -124,9 +124,9 @@ struct TiDBSchemaSyncer : public SchemaSyncer
         if (version_after_load_diff = tryLoadSchemaDiffs(getter, version, context); version_after_load_diff == -1)
         {
             GET_METRIC(tiflash_schema_apply_count, type_full).Increment();
-            loadAllSchema(getter, version, context);
-            // After loadAllSchema, we need update `version_after_load_diff` by last diff value exist or not
+            // we need update `version_after_load_diff` by last diff value exist or not
             version_after_load_diff = getter.checkSchemaDiffExists(version) ? version : version - 1;
+            loadAllSchema(getter, version_after_load_diff, context);
```

Comment:
I think making it to be
`version_after_load_diff = loadAllSchema(getter, version, context);`
and move `getter.checkSchemaDiffExists` into the function `loadAllSchema` is better.

After that change, the two functions `tryLoadSchemaDiffs`, `loadAllSchema` handle similar problem inside the function